### PR TITLE
Newsletter vol.4

### PR DESCRIPTION
**Pull request by syleeeee**
- Add the newsletter vol. 4

Discussion items
- Should we create some more good first issues? [Currently, all of them are WIP](https://github.com/line/armeria/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22).
- I just added all of the release note summaries, but should we merge and display them in small groups? Such as `New support`, `gRPC new features`, `Fixes`, etc ..

Any other comments are appreciated :) 

screencapture-localhost-8000-news-20220708-newsletter-4-2022-07-07-17_13_40

### CODE REVIEW

File: site/src/pages/news/20220708-newsletter-4.mdx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+<div class="noBullet">
+
+- 🎓 Try out this guided tutorial for REST Service](https://armeria.dev/tutorials/rest/blog)! It's super simple, and super friendly to understand building a RESTful service with Aremria. This tutorial will guide you to make a blog service, with which you can create, read, update and delete blog posts. In particular, you'll be using these Armeria features:
+  - Service annotations
+  - Request converter
+  - Parameter injection
+  - Exception handler
+  - Blocking task executor
+- 🦾 MSA(Microservices Architecture): Harder, Better, Faster, Stronger
+  - In the session, @cnabro introduces how the "Series On" service converted monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentioned about how they monitored microservices using Zipkin and Elastic Search.
```
Comment (minwoox) [inline site/src/pages/news/20220708-newsletter-4.mdx:59]:
```suggestion
  - In the session, @cnabro introduces how [Series On](https://serieson.naver.com/) migrated their monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentioned how they monitored microservices using [Zipkin](https://zipkin.io/) and [Elasticsearch](https://www.elastic.co/).
```

**Comment by minwoox** [comment]
> Should we create some more good first issues?

Let us create more issues before we send this letter. 😉 

> I just added all of the release note summaries, but should we merge and display them in small groups?

I prefer the current way except for putting the latest release at the top. 😄

File: site/src/pages/news/20220708-newsletter-4.mdx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    - Easier way to send and receive RESTful APIs
+    - Ability to limit the maximum length of Thrift messages
+    - Injecting dependencies in annotations
+    - Support for Kotlin nullable type(`?`) in annotated services
+
+</div>
+<div class="noBullet">
+
+- 👂 Armeria team is hosting [bi-weekly virtual office hours](https://armeria.dev/community/office-hours/).
+  - It was great talking with all of you through the video chat. And we're still listening. :)
```
Comment (jrhee17) [inline site/src/pages/news/20220708-newsletter-4.mdx:44]: I think the sentence sounds a little vague. Did you mean we're still doing virtual hours?

```suggestion
  - It was great talking with all of you through video chat. We hope we can see you again in the near future. :)
```

File: site/src/pages/news/20220708-newsletter-4.mdx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+<video class="hideOnReducedMotion" src="../../images/armeria.m4v" preload="none" autoplay muted loop style="width: 282px; height: 112px;">
+  <img src="../../images/armeria.gif" loading="lazy" width="282" height="112" />
+</video>
+
+## From the devs
+
+<div class="noBullet">
+
+- ✨ We released several releases since our last newsletter with a bunch of new exciting features!
```
Comment (ikhoon) [inline site/src/pages/news/20220708-newsletter-4.mdx:15]: According to [Amazon's writing style tips](https://medium.com/fact-of-the-day-1/amazon-writing-style-tip-2-2f77bb3bbbb0), it recommends replacing adjectives with numbers. 😆
```suggestion
- ✨ We've released 7 new versions since our last newsletter with a bunch of new exciting features!
```

File: site/src/pages/news/20220708-newsletter-4.mdx (hunk 1 above)
Comment (jrhee17) [inline site/src/pages/news/20220708-newsletter-4.mdx:59]: The tense should be consistent (introduces <-> mentions)

```suggestion
  - In the session, @cnabro introduces how the "Series On" service converted monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentions how they monitored microservices using Zipkin and Elastic Search.
```

File: site/src/pages/news/20220708-newsletter-4.mdx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+- 👂 Armeria team is hosting [bi-weekly virtual office hours](https://armeria.dev/community/office-hours/).
+  - It was great talking with all of you through the video chat. And we're still listening. :)
+
+</div>
+
+## From the community
+
+<div class="noBullet">
+
+- 🎓 Try out this guided tutorial for REST Service](https://armeria.dev/tutorials/rest/blog)! It's super simple, and super friendly to understand building a RESTful service with Aremria. This tutorial will guide you to make a blog service, with which you can create, read, update and delete blog posts. In particular, you'll be using these Armeria features:
```
Comment (jrhee17) [inline site/src/pages/news/20220708-newsletter-4.mdx:52]: - Shouldn't the link be on `tutorial` instead of `REST Service`?
- Prefer active voice, "It" is vague in this context
- Removed a redundant sentence

```suggestion
- 🎓 Try out this guided [tutorial](https://armeria.dev/tutorials/rest/blog) for creating a RESTful service! This super friendly and simple tutorial will guide you to make a blog service, with which you can create, read, update and delete blog posts. In particular, you'll be using these Armeria features:
```

File: site/src/pages/news/20220708-newsletter-4.mdx [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+  - Service annotations
+  - Request converter
+  - Parameter injection
+  - Exception handler
+  - Blocking task executor
+- 🦾 MSA(Microservices Architecture): Harder, Better, Faster, Stronger
+  - In the session, @cnabro introduces how the "Series On" service converted monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentioned about how they monitored microservices using Zipkin and Elastic Search.
+  - Here are the [Video](https://youtu.be/1TsGYR1dves) and [Slide](https://speakerdeck.com/line_devday2021/msa-microservices-architecture-harder-better-faster-stronger) (also providing [Korean](https://youtu.be/95L-_82N1vg) and [Japanese](https://youtu.be/pXIdcgchvSU))
```
Comment (jrhee17) [inline site/src/pages/news/20220708-newsletter-4.mdx:60]: Since the subject "Here" is singular

```suggestion
  - Here is the [Video](https://youtu.be/1TsGYR1dves) and [Slide](https://speakerdeck.com/line_devday2021/msa-microservices-architecture-harder-better-faster-stronger) (also providing [Korean](https://youtu.be/95L-_82N1vg) and [Japanese](https://youtu.be/pXIdcgchvSU))
```

File: site/src/pages/news/20220708-newsletter-4.mdx [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+  - [1.14.0](https://armeria.dev/release-notes/1.14.0/)
+    - Experimental support for Scala 3
+    - Fluent conversion of <type://HttpResponse> into a desired type using <type://WebClient>
+    - Various build options for gRPC using <type://GrpcClientBuilder>
+  - [1.14.1](https://armeria.dev/release-notes/1.14.1/)
+    - Additional logs for <type://EmptyEndpointGroupException>
+  - [1.15.0](https://armeria.dev/release-notes/1.15.0/)
+    - User-friendly gRPC health check protocol
+    - Multipart file uploads in annotated services
+    - Preliminary GraphQL support in <type://DocService>
```
Comment (jrhee17) [inline site/src/pages/news/20220708-newsletter-4.mdx:27]: Since preliminary sounds like it's not ready for usage, but it is 😄 

```suggestion
    - GraphQL support in <type://DocService>
```

File: site/src/pages/news/20220708-newsletter-4.mdx [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    - Refactored DMS resolver cache with higher hit ratio and reduced traffic
```
Comment (jrhee17) [inline site/src/pages/news/20220708-newsletter-4.mdx:28]:
```suggestion
    - Refactored DNS resolver cache with higher hit ratio and reduced traffic
```

File: site/src/pages/news/20220708-newsletter-4.mdx [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+  - [1.13.4](https://armeria.dev/release-notes/1.13.4/)
+    - Security fix(CWE-22)
+  - [1.14.0](https://armeria.dev/release-notes/1.14.0/)
+    - Experimental support for Scala 3
+    - Fluent conversion of <type://HttpResponse> into a desired type using <type://WebClient>
+    - Various build options for gRPC using <type://GrpcClientBuilder>
+  - [1.14.1](https://armeria.dev/release-notes/1.14.1/)
+    - Additional logs for <type://EmptyEndpointGroupException>
+  - [1.15.0](https://armeria.dev/release-notes/1.15.0/)
+    - User-friendly gRPC health check protocol
```
Comment (jrhee17) [inline site/src/pages/news/20220708-newsletter-4.mdx:25]: I'm not sure if this additional support is particularly use friendly

```suggestion
    - Support for gRPC health check protocol
```

**Review by jrhee17 — COMMENTED** [review]
Overall looks good 👍  Left some minor comments

**Comment by syleeeee** [comment]
Thanks for reviewing. I addressed your comments.
screencapture-localhost-8000-news-20220714-newsletter-4-2022-07-14-16_24_43

File: site/src/pages/news/20220714-newsletter-4.mdx [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+  - #4237 Support for request scoped DataLoader for GraphQL-Java
+  ```java
+  class GraphqlServiceBuilder {
+      public GraphqlServiceBuilder dataLoaderRegistry(
+        Function<ServiceRequestContext,
+        DataLoaderRegistry>dataLoaderRegistrySupplier) {
+           // This method will be mutually exclusive
+           // with configureDataLoaderRegistry(...)
+      }
+  }
```
Comment (ikhoon) [inline site/src/pages/news/20220714-newsletter-4.mdx:91]:
```suggestion
  class GraphqlServiceBuilder {
    public GraphqlServiceBuilder dataLoaderRegistry(
      Function<ServiceRequestContext,
        DataLoaderRegistry> dataLoaderRegistrySupplier) {
          // This method will be mutually exclusive
          // with configureDataLoaderRegistry(...)
    }
  }
```

File: site/src/pages/news/20220714-newsletter-4.mdx [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+  ```
+  - #4347 Show the full stack trace of <type://CompositeException> when the system property verboseExceptions is `true`
```
Comment (jrhee17) [inline site/src/pages/news/20220714-newsletter-4.mdx:93]:
```suggestion
  - #4347 Show the full stack trace of <type://CompositeException> when the system property `verboseExceptions` is `true`
```

File: site/src/pages/news/20220714-newsletter-4.mdx [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+
+- 🎓 Try out this guided [tutorial](https://armeria.dev/tutorials/rest/blog) for REST Service! This super friendly and simple tutorial will guide you to make a blog service, with which you can create, read, update and delete blog posts. In particular, you'll be using these Armeria features:
+  - Service annotations
+  - Request converter
+  - Parameter injection
+  - Exception handler
+  - Blocking task executor
+- 🦾 MSA(Microservices Architecture): Harder, Better, Faster, Stronger
+  - In the session, @cnabro introduces how [Series On](https://serieson.naver.com/) migrated their monoliths into microservices using Armeria, gRPC, and GraphQL. He also mentions about how they monitored microservices using [Zipkin](https://zipkin.io/) and [Elasticsearch](https://www.elastic.co/).
+  - Here is the [Video](https://youtu.be/1TsGYR1dves) and [Slide](https://speakerdeck.com/line_devday2021/msa-microservices-architecture-harder-better-faster-stronger) (also providing [Korean](https://youtu.be/95L-_82N1vg) and [Japanese](https://youtu.be/pXIdcgchvSU))
```
Comment (jrhee17) [inline site/src/pages/news/20220714-newsletter-4.mdx:62]: Oops, I didn't see the entire sentence correctly 😅 Let me revert this change for you.

```suggestion
  - Here are the [Video](https://youtu.be/1TsGYR1dves) and [Slides](https://speakerdeck.com/line_devday2021/msa-microservices-architecture-harder-better-faster-stronger) (also providing [Korean](https://youtu.be/95L-_82N1vg) and [Japanese](https://youtu.be/pXIdcgchvSU))
```

Comment (syleeeee) [reply site/src/pages/news/20220714-newsletter-4.mdx]: I trusted you more than I trusted myself 🤣

File: site/src/pages/news/20220714-newsletter-4.mdx [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+  <img src="../../images/armeria.gif" loading="lazy" width="282" height="112" />
+</video>
+
+## From the devs
+
+<div class="noBullet">
+
+- ✨ We've released 7 new versions since our last newsletter with a bunch of new exciting features!
+  - [1.17.1](https://armeria.dev/release-notes/1.17.1/)
+    - Fixing a bug where Retrofit client calls hang indefinitely
```
Comment (trustin) [inline site/src/pages/news/20220714-newsletter-4.mdx:17]: How about removing micro versions in the list, so we focus on new features only? I'm afraid people might stop reading if the list gets too large.

File: site/src/pages/news/20220714-newsletter-4.mdx [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+## From the devs
+
+<div class="noBullet">
+
+- ✨ We've released several new versions since our last newsletter with a bunch of new exciting features!
+  - [1.17.0](https://armeria.dev/release-notes/1.17.0/)
+    - Easier way to send and receive RESTful APIs
+    - Ability to limit the maximum length of Thrift messages
+    - Injecting dependencies in annotations
+    - Support for Kotlin nullable type(`?`) in annotated services
```
Comment (trustin) [inline site/src/pages/news/20220714-newsletter-4.mdx:20]:
```suggestion
    - Support for Kotlin nullable type (`?`) in annotated services
```

**Review by trustin — APPROVED** [review]
Just one nit - Thanks a lot, @syleeeee! 🙇

**Comment by syleeeee** [comment]
Thanks all for reviewing :-) 
Newsletter vol.4 is published.

**Comment by ikhoon** [comment]
I've received it! 😆 Thanks a lot!
